samza-test/src/test/java/org/apache/samza/test/operator/RepartitionJoinWindowApp2.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.test.operator;
+
+import org.apache.samza.application.StreamApplication;
+import org.apache.samza.config.Config;
+import org.apache.samza.operators.KV;
+import org.apache.samza.operators.MessageStream;
+import org.apache.samza.operators.StreamGraph;
+import org.apache.samza.operators.functions.JoinFunction;
+import org.apache.samza.operators.windows.Windows;
+import org.apache.samza.serializers.JsonSerdeV2;
+import org.apache.samza.serializers.KVSerde;
+import org.apache.samza.serializers.StringSerde;
+import org.apache.samza.system.OutgoingMessageEnvelope;
+import org.apache.samza.system.SystemStream;
+import org.apache.samza.task.TaskCoordinator;
+import org.apache.samza.test.operator.data.AdClick;
+import org.apache.samza.test.operator.data.PageView;
+import org.apache.samza.test.operator.data.UserPageAdClick;
+
+import java.time.Duration;
+
+/**
+ * A {@link StreamApplication} that demonstrates a partitionBy, stream-stream join and a windowed count.
+ */
+public class RepartitionJoinWindowApp2 implements StreamApplication {
```

Comment:
Why copy this? If this is meant to be a separate test maybe rename it?

Reply:
`RepartitionJoinWindowApp` is used by `testRepartitionJoinWindowAppAndDeleteMessagesOnCommit()` and `RepartitionJoinWindowApp2` is used by `testRepartitionJoinWindowAppWithoutDeletionOnCommit()`. `RepartitionJoinWindowApp2` is almost the same as `RepartitionJoinWindowApp` except for the topic names used in the app. If these two unit tests use the same app with the same Kafka topics, the second test will fail. I am not able to get around this problem. That is why I added `RepartitionJoinWindowApp2`.

Reply:
Possible to make stream names constructor params for that class in that case?

Reply:
@prateekm Sure. I have updated the patch as suggested.